pkg/minishift/network/settings.go

```diff
@@ -43,31 +57,254 @@ type NetworkSettings struct {
+	}
+
+	printNetworkSettings(networkSettings)
+
+	fmt.Println(configureRestartNeededMessage)
 }
 
+// printNetworkSettings will print to stdout the values from the struct
+// NetworkSettings.
 func printNetworkSettings(networkSettings NetworkSettings) {
 	fmt.Println(configureIPAddressMessage)
 	fmt.Println("   Device:     ", networkSettings.Device)
 	fmt.Println("   IP Address: ", fmt.Sprintf("%s/%s", networkSettings.IPAddress, networkSettings.Netmask))
 	if networkSettings.Gateway != "" {
 		fmt.Println("   Gateway:    ", networkSettings.Gateway)
 	}
-	if networkSettings.DNS1 != "" {
+	if networkSettings.DNS1 != "" || networkSettings.DNS2 != "" {
 		fmt.Println("   Nameservers:", fmt.Sprintf("%s %s", networkSettings.DNS1, networkSettings.DNS2))
 	}
 }
 
+// fillNetworkSettings will populate the network configuration file for use
+// by the minishift-set-ipaddress script inside the VM. It takes the struct
+// NetworkSettings containing the values and retuns a string containing
+// a multiline config file.
 func fillNetworkSettingsScript(networkSettings NetworkSettings) string {
 	result := &bytes.Buffer{}
 
-	tmpl, err := template.New("networkScript").Parse(configureNetworkScriptTemplate)
-	if err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating network script template: %s", err.Error()))
+	tmpl := template.New("networkScript")
+
+	if networkSettings.Disabled {
+		tmpl.Parse(configureNetworkScriptDisabledAddressTemplate)
+	} else if networkSettings.UseDHCP {
+		tmpl.Parse(configureNetworkScriptDynamicAddressTemplate)
+	} else {
+		tmpl.Parse(configureNetworkScriptStaticAddressTemplate)
 	}
-	err = tmpl.Execute(result, networkSettings)
+
+	err := tmpl.Execute(result, networkSettings)
 	if err != nil {
-		atexit.ExitWithMessage(1, fmt.Sprintf("Error executing network script template:: %s", err.Error()))
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error executing network script template: %s", err.Error()))
 	}
 
 	return result.String()
 }
+
+func executeCommandOrExit(driver drivers.Driver, command string, errorMessage string) string {
+	result, err := drivers.RunSSHCommandFromDriver(driver, command)
+
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("%s: %s", errorMessage, err.Error()))
+	}
+	return result
+}
+
+// HasNameserversConfigured returns true if the instance uses nameservers
+// This is related to an issues when LCOW is used on Windows.
+func HasNameserversConfigured(driver drivers.Driver) bool {
+	cmd := "cat /etc/resolv.conf | grep -i '^nameserver' | wc -l | tr -d '\n'"
+	out, err := drivers.RunSSHCommandFromDriver(driver, cmd)
+
+	if err != nil {
+		return false
+	}
+
+	i, _ := strconv.Atoi(out)
+
+	return i != 0
+}
+
+// AddNameserversToInstance will add additional nameservers to the end of the
+// /etc/resolv.conf file inside the instance.
+func AddNameserversToInstance(driver drivers.Driver, nameservers []string) {
+	// TODO: verify values to be valid
```

Comment:
@gbraad , I see TODO here. Is it not done or you forgot to remove it?

Reply:
We need to implement reachability and validness checks, like check if poet 53 listens, etc. Not something I will implement now. New issue will be filed afterwards.

I wonder why you are so against having TODO placeholders

Reply:
I will prefer issues for future TODOs. IMO that a better way to track technical debt.